files/rconcmds/rcon-cmds-daemon.sh

```diff
@@ -52,19 +54,34 @@ do
     fi
     ;;
   XII)
-    # Main Loop looking for connections 
     CURR_CLIENTCONNECTIONS=$(java_clients_connections)
+    # When a client joins
     if (( CURR_CLIENTCONNECTIONS > CLIENTCONNECTIONS )) && [[ "$RCON_CMDS_ON_CONNECT" ]]; then
         logRcon "Clients have Connected, running connect cmds"
         while read -r cmd; do
           run_command "$cmd"
         done <<< "$RCON_CMDS_ON_CONNECT"
+    # When a client leaves
     elif (( CURR_CLIENTCONNECTIONS < CLIENTCONNECTIONS )) && [[ "$RCON_CMDS_ON_DISCONNECT" ]]; then
         logRcon "Clients have Disconnected, running disconnect cmds"
         while read -r cmd; do
           run_command "$cmd"
         done <<< "$RCON_CMDS_ON_DISCONNECT"
     fi
+
+    # First client connection
+    if (( CURR_CLIENTCONNECTIONS > 0 )) && (( CLIENTCONNECTIONS == 0 )) && [[ "$RCON_CMDS_FIRST_CONNECT" ]]; then
```

Comment:
Up on line 49 I think you'll also need to consider these two new variables when deciding the `exit 0` early.